sonar-text-plugin/src/main/java/org/sonar/plugins/secrets/configuration/model/matching/Detection.java

```diff
@@ -20,27 +20,47 @@
 
 package org.sonar.plugins.secrets.configuration.model.matching;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
 import javax.annotation.Nullable;
+import org.sonar.plugins.secrets.configuration.model.Rule;
 import org.sonar.plugins.secrets.configuration.model.matching.filter.PostModule;
 import org.sonar.plugins.secrets.configuration.model.matching.filter.PreModule;
 
 public class Detection {
+
+  @Nullable
   private Match matching;
   @Nullable
   private PreModule pre;
   @Nullable
   private PostModule post;
+  @JsonIgnore
+  private Rule rule;
+
+  private boolean associatedProviderDetectionExists() {
+    if (rule != null && rule.getProvider() != null) {
+      return rule.getProvider().getDetection() != null;
+    }
+    return false;
+  }
 
+  @Nullable
   public Match getMatching() {
+    if (matching == null && associatedProviderDetectionExists()) {
+      return rule.getProvider().getDetection().getMatching();
```

Comment:
`getMatching` can return null since you allow the argument of `setMatching` to be nullable (see next lines)